### ISSUE TITLE
Authentication: Failed to instantiate AWSMobileClient

### ISSUE THREAD

**Issue by infernal008** [aws-amplify/amplify-flutter#508]
I have recently added **auth** through the Amplify CLI by importing an existing Cognito User Pool and Identity Pool.

So what happens to me is that upon starting my app on the emulator, I am blocked by an AmplifyException error like below:

> Exception has occurred.
> AmplifyException (AmplifyException(message: Failed to instantiate AWSMobileClient, recoverySuggestion: See attached exception for more details, underlyingException: java.lang.RuntimeException: Failed to initialize OAuth, please check your awsconfiguration.json))

Here's a summary of my observations:

1. I was actually able to make this work before though on different plugin versions
2. I don't have any **awsconfiguration.json** file anywhere, as referenced by the exception message above
3. I have only noticed this problem occurring after updating my plugins to ^0.1.1
4. I'm not sure if it could be a configuration problem or something else
5. I have followed the tutorial even up to updating my AndroidManifest.xml file
6. I have been able to make _sign-in with hosted ui_ work before with the same configuration (but different plugin versions)
7. The exception is pointed when executing _Amplify.configure([configObject])_
8. The app would work fine whenever I remove the Auth Cognito plugin
9. Though blocked by the exception above on the first run, I can see the following logs below:

> I/amplify:flutter:auth_cognito( 9830): Added Auth plugin
> D/AWSMobileClient( 9830): initialize: Cognito HostedUI client detected
> D/AWSMobileClient( 9830): Inspecting user state details
> W/InternalCallback( 9830): Library attempted to call user callback twice, expected only once

Below is my configuration file for reference:

```
const amplifyconfig = ''' {
    "UserAgent": "aws-amplify-cli/2.0",
    "Version": "1.0",
    "api": {
        "plugins": {
            "awsAPIPlugin": {
                "[API NAME]": {
                    "endpointType": "GraphQL",
                    "region": "[REGION]",
                    "authorizationType": "[AUTHORIZATION TYPE]"
                }
            }
        }
    },
    "auth": {
        "plugins": {
            "awsCognitoAuthPlugin": {
                "UserAgent": "aws-amplify-cli/0.1.0",
                "Version": "0.1.0",
                "IdentityManager": {
                    "Default": {}
                },
                "CredentialsProvider": {
                    "CognitoIdentity": {
                        "Default": {
                            "PoolId": "[POOL ID]",
                            "Region": "[REGION]"
                        }
                    }
                },
                "CognitoUserPool": {
                    "Default": {
                        "PoolId": "[POOL ID]",
                        "AppClientId": "[APP CLIENT ID]",
                        "Region": "[REGION]"
                    }
                },
                "Auth": {
                    "Default": {
                        "OAuth": {
                            "WebDomain": "[WEB DOMAIN]",
                            "AppClientId": "[APP CLIENT ID]",
                            "SignInRedirectURI": "[URI]",
                            "SignOutRedirectURI": "[URI]",
                            "Scopes": [
                                "aws.cognito.signin.user.admin",
                                "email",
                                "openid",
                                "phone",
                                "profile"
                            ]
                        },
                        "authenticationFlowType": "USER_SRP_AUTH"
                    }
                }
            }
        }
    }
}''';
```

Here's some information about my working environment:

Flutter 2.0.2
Dart 2.12.1
amplify_core: ^0.1.1
amplify_flutter: ^0.1.1
amplify_auth_cognito: ^0.1.1
amplify_datastore: ^0.1.1
amplify_datastore_plugin_interface: ^0.1.1
amplify_api: ^0.1.1
Amplify CLI 4.46.1

**Comment by infernal008** [comment]
I would just want to update my filing here with what had happend to me so far. Basically, I got it to work now.

I tried commenting and uncommenting the activity tag and attributes that were instructed to be added by the Auth Documentation and after running it, I am no longer stuck in the exception that I have reported above and am now able to successfully log in and log out of my account.

Nothing else were added or removed, it seems that vscode just have some config or files that were cached and when i tried shuffling them, it would seem to use the correct ones after.

I'll try to help if I could if anything else would be needed for this filing. Please just inform me if this is to be closed or not yet for further investigation.

**Comment by haverchuck** [comment]
@infernal008 Thanks for updating - it seems like we can close this for now if you are unblocked.

**Comment by amkuchta** [comment]
@haverchuck I am having the same issue as above - I tried the "comment / uncomment" fix, and didn't have any luck.  Should I open a new ticket, or can we re-open this one?  For reference, my config is below:


### Error Message
> PlatformException (PlatformException(AmplifyException, Translating platform exception failed. Please take a look at
https://github.com/aws-amplify/amplify-flutter/issues to see if there are any existing issues that
match your scenario, and file an issue with the details of the bug if there isn't., {recoverySuggestion: See attached exception for more details, underlyingException: java.lang.RuntimeException: Failed to initialize OAuth, please check your awsconfiguration.json, message: Failed to instantiate AWSMobileClient}, null))

### Config
_**Note: Sensitive data redacted, replaced with [ITEM_NAME]**_
```dart
{
  "UserAgent": "aws-amplify-cli/2.0",
  "Version": "1.0",
  "auth": {
    "plugins": {
      "awsCognitoAuthPlugin": {
        "IdentityManager": {
          "Default": {}
        },
        "CredentialsProvider": {
          "CognitoIdentity": {
            "Default": {
              "PoolId": "[IDENTITY_POOL_ID]",
              "Region": "us-east-1"
            }
          }
        },
        "CognitoUserPool": {
          "Default": {
            "PoolId": "[USER_POOL_ID]",
            "AppClientId": "[CLIENT_ID]",
            "Region": "us-east-1"
          }
        },
        "Auth": {
          "Default": {
            "OAuth": {
              "WebDomain": "https://dev.auth.[DOMAIN].com",
              "AppClientId": "[APP_CLIENT_ID]",
              "SignInRedirectURI": "[URI]://",
              "SignOutRedirectURI": "[URI]://",
              "Scopes": [
                "phone",
                "email",
                "openid",
                "profile",
                "aws.cognito.signin.user.admin"
              ]
            },
            "authenticationFlowType": "USER_SRP_AUTH",
            "loginMechanism": ["EMAIL"],
            "signupAttributes": ["EMAIL"],
            "passwordProtectionSettings": {
              "passwordPolicyMinLength": 8,
              "passwordPolicyCharacters": []
            },
            "mfaConfiguration": "OFF",
            "mfaTypes": ["SMS"]
          }
        }
      }
    }
  },
  "api": {
    "plugins": {
      "awsAPIPlugin": {
        "Dev-[APP_NAME]-GraphQL-API": {
          "endpointType": "GraphQL",
          "endpoint": "https://[ENDPOINT].appsync-api.us-east-1.amazonaws.com/graphql",
          "region": "us-east-1",
          "authorizationType": "AMAZON_COGNITO_USER_POOLS"
        }
      }
    }
  }
}
```

### Environment Info
**Flutter Version:** 2.0.4
**Dart Version:** 2.12.2
**Amplify Version:** 5.3.0
**amplify_flutter Version:** ^0.2.2
**amplify_auth_cognito Version:** ^0.2.2
**amplify_api Version:** ^0.2.2
**amplify_storage_s3 Version:** ^0.2.2

**Comment by Joao-VictorF** [comment]
Hey @amkuchta ! I'm having the same problem, have you found a workaround for this issue?
If you remeber the solution, can you share with me, please?

**Comment by sumanthns-commits** [comment]
I am facing the same issue. Are they any more updates on this?

**Comment by Joao-VictorF** [comment]
I solved this a few days ago, if I remeber well, I did this steps:

1. Comment the changes in AndroidManifest & stop and run the app
2. Uncomment and stop and run the app again
3. if the error still persists, completely uninstall de app in your device and run the build again.

**Comment by fuzzy28** [comment]
This issue shouldn't be closed as people are facing this

**Comment by dnys1** [comment]
Hi @fuzzy28 I have reopened the issue. If you'd be willing, please provide the following information and I'll be happy to investigate:

- Amplify CLI version
- Amplify Flutter library version
- Android version testing on
- `amplifyconfiguration.dart` (can be redacted)

Thank you and sorry you are still experiencing this issue.

**Comment by Jordan-Nelson** [comment]
@fuzzy28 - If you are still experiencing this issue can you please provide the info requested above?

**Comment by Jordan-Nelson** [comment]
Hey everyone - I am going to close this out since we have not heard back in some time. If anyone is still experiencing this issue, please open a new issue with the info request above so that we can attempt to reproduce this.
